### PR TITLE
Headless unit tests script

### PR DESCRIPTION
**Pull request by dmohs**
Allows headless unit testing for pre-merge checks.

https://broadinstitute.atlassian.net/browse/GAWB-1539

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [x] **LR** sign off
- [x] **Product Owner** sign off
- [x] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: scripts/test.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+compose_exec clojure lein with-profile deploy npm install
+compose_exec clojure ./node_modules/webpack/bin/webpack.js -p
+compose_exec clojure lein cljsbuild once
+compose_exec clojure lein resource
+
+docker cp scripts/.phantom-run-tests.js fcuitests_clojure_1:/w/run-tests.js
+compose_exec phantomjs phantomjs run-tests.js
+
+clean_up
```
Comment (zarsky-broad) [inline scripts/test.sh]: I think `clean_up` is getting called twice. Here's my console output at the end of running this script:

```
Removing network fcuitests_default
WARNING: Network fcuitests_default not found.
Volume node-modules is external, skipping
Removing volume fcuitests_work
WARNING: Volume fcuitests_work not found.
```

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+services:
+  - docker
+
+sudo: false
+
+script:
+  - scripts/test.sh
+
+# This isn't used, but makes slightly more sense than ruby, the default.
+language: java
```
Comment (zarsky-broad) [inline .travis.yml]: If it isn't used, could we just not specify it? I feel like fewer keys is just easier.

File: scripts/.phantom-run-tests.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    } else {
+      console.log(msg);
+    }
+  };
+}
+
+page.open('http://server/', function(status) {
+  installConsoleListener();
+  page.evaluate(function() {
+    var logToken = 'e54b0a2b-0934-400f-bf3a-68d01dcde2df';
```
Comment (zarsky-broad) [inline scripts/.phantom-run-tests.js]: Why redefine this?

File: scripts/.phantom-run-tests.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+var page = require('webpage').create();
```
Comment (zarsky-broad) [inline scripts/.phantom-run-tests.js:1]: Any reason not to drag this into the future? Make this and `logToken` into `const`s, `let someTestsFailed`...

File: .travis.yml (hunk 2 above)
Comment (dmohs) [reply .travis.yml]: I just think it makes more sense to run these tests in a Java environment vs. a Ruby environment, even though everything will actually happen in a Docker environment. But yeah probably better to just let it be Ruby.

File: scripts/.phantom-run-tests.js (hunk 3 above)
Comment (dmohs) [reply scripts/.phantom-run-tests.js]: It's not redefined. The first one is within the execution environment, the second is within the browser environment. I think making both `const`s will help clarify that.

Comment (zarsky-broad) [reply scripts/.phantom-run-tests.js]: Let me know if I'm confused, but take a look at the scoping. The previous definition is at the root scope of this file. It should be available here.

File: scripts/.phantom-run-tests.js (hunk 4 above)
Comment (dmohs) [reply scripts/.phantom-run-tests.js:1]: Does this commit [1] convince you it is a different execution environment?
[1] daba905

**Review by zarsky-broad — APPROVED** [review]

**Pull request merged**